### ISSUE TITLE
Long product codes are truncated

### ISSUE THREAD

**Issue by JukePlz** [ShendoXT/memcardrex#35]
Using the BIOS built-in SoundScope function to record a macro, and then saving the macro to memory card produces a save file with the product code "BPLAYSTATION" but MemcardRex only shows "LAYSTATION" and the header editor doesn't support editing text longer than 10 characters. Proper behaviour should support displaying and editing text to at least 12 characters instead.

Region also shows up as "0x5042" could probably add an human readable name for the region, either as World or BIOS, but more research should be done to determine if that specific region code is used on all BIOS versions or if any other game or software uses that region code.

Attached is an example MMC with a SoundScope save.

[soundscope.zip](https://github.com/ShendoXT/memcardrex/files/9126965/soundscope.zip)

**Comment by ShendoXT** [comment]
Well, here is the deal. The region+product code+identifier is actually a raw save file name on the file system, a 20 character string.
Sony forced developers to use the naming scheme in this format but they themselves as you can see were not following
this standard. Realistically I could allow editing of the whole string but ever since the dawn of time editors were doing it this way... I could add an option in preferences to always raw edit the header I guess.

**Comment by ShendoXT** [comment]
<img width="523" alt="Screenshot 2023-11-29 at 09 41 19" src="https://github.com/ShendoXT/memcardrex/assets/8411572/8609d02e-26f7-4c30-92d8-0bc2e73cdcdf">

What do you think about this solution?
Editor is lying about product code but header editor is revealing the truth.

**Comment by ShendoXT** [comment]
Yes, all the changes will be ported back to Windows version too.
